### ISSUE TITLE
[Request New Content] Better Machine Learning Demands Better Data Labeling

### ISSUE THREAD

**Issue by theodu** [kili-technology/kili-blogger-program#13]
### My Request

Better Machine Learning Demands Better Data Labeling

### Community vote
👍 Upvote if you like this topic
🚀 If you can make the article/video!

Thank you.

**Comment by EdemGold** [comment]
The performance of Machine Learning models depends entirely(among other things) on how well the data it is being trained with is labeled.

OUTLINE FOR PROPOSED ARTICLE
- Explain what data labeling is
- Explain how data labeling works.
- Explain why data labeling is important

**Comment by joybh98** [comment]
Hi @theodu , I'm interested in this article ! Here is my approach (i'm flexible about the approach)

Data labelling is a very important part of the preprocessing stage when developing machine learning models 

Outline:

1. How data labelling works(human interaction, accuracy)
2. Approaches
3. Difference in performance of ML models with labelled and unlabelled data(only applicable where unlabeled can be used)
4. Best practices

**Comment by theodu** [comment]
Hi @EdemGold and @joybh98 
Thank you for your outlines! Could you please provide a more detailed one? It is hard to visualize what you have in mind.

Also, I think that showing that improving the data can improve the model performance through a concrete example could be very important. We tend to say it a lot but illustrating it through an example with one same model and two slightly different datasets could be very demonstrative. Do you think that you could integrate such an example in your article?

**Comment by joybh98** [comment]
> Also, I think that showing that improving the data can improve the model performance through a concrete example could be very important. We tend to say it a lot but illustrating it through an example with one same model and two slightly different datasets could be very demonstrative. Do you think that you could integrate such an example in your article?

I completely agree @theodu , I've also included this in the points that I will add to the article. I'll be glad to integrate such an example.

>Thank you for your outlines! Could you please provide a more detailed one? It is hard to visualize what you have in mind.

1. How data labelling works 

- When working with an ML/DL/AI model, most of the times you'll start with a lot of unlabeled data. Data labelling is the process of detecting and tagging data samples/dataset. Data labelling workflow often includes data annotation,tagging, classification, moderation and processing. 

2. Data Labelling Approaches
- In-house
- Outsourcing
- Crowdsourcing
- Synthetic
- Programmed

NOTE: I'll give examples for each of these approaches 

3. Difference in performance of ML models with labelled and unlabelled data(only applicable where unlabeled can be used)
- I have yet to decide on which type of model I'm going to use to show the difference, but I was thinking I can use Kili annotation products for an example ?

4. Best practices
- Collect diverse amount of data
- Establish a QA process
- Choose a tagging taxonomy that's unique to your work/business/research etc


As always, I'm open to suggestions, and if you want to remove/add something please feel free to ask.

**Comment by joybh98** [comment]
hi @theodu , I know you're a busy person,but I would appreciate a reply whenever you're free :)

**Comment by theodu** [comment]
Hi @joybh98 
Sorry for the late reply. It was more the fact that it was the end-of-year period that freezes operations than the fact that I am busy ;)
I really like this outline, I think that you will do a great job. I assign you to the article, you can begin to work on it!

**Comment by joybh98** [comment]
@theodu I'm happy to discuss what all do you want from the article? 

> I find this idea in your 3) but the whole article should be oriented around this idea.

I agree, I wanted to give a broad overview about data annotation and what all it encompasses but I understand how it can be redundant. 

Should we continue the discussion here or somewhere else ?

**Comment by theodu** [comment]
@joybh98 the issue indeed lacks a description to know what we had in mind behind this title. From now on, I will always add a description when I propose an article.
I already gave you some ideas in the comments. As I said, the core of the article should be a concrete illustration/proof through an example that two differently labeled datasets (one with some errors, unfitted labels, etc...) can lead to a big performance difference (even with only a few errors). 
The easiest is if you can propose a new outline based on these comments, for busy calendar reasons, I prefer not to book a meeting if that is possible. But I am happy to answer you by mail: theo.dullin@kili-technology.com

Best!
Théo

**Comment by Renerd-max** [comment]
Hi, @theodu, here is my outline.

**Introduction:** An introduction to machine learning, and its challenges, which involves building a better AI model, towards the end of the introduction, I will summarise how this challenge can be effectively handled by focusing on the data used in building the AI model, data remains the most important aspect of machine learning.

**Sub-Topic 1: Better machine learning demands better data labeling (Case studies)**

**Case Study 1:** Case study 1 will explain the process of building an AI model, let say a model to recognize a human face, the data will contain some errors, and the relationship between the error and the result will be analyzed.

**Case Study 2: ** Case study 2 explains the process of building another AI, but the annotators are careful with the data, and all due diligence has been taken by the project manager to ensure the best quality of data, this results in a perfect mode.

I will round up by comparing the two models and the importance of quality data in building quality machine learning models.

**Sub-Topic 2: How Kili helps label better Data**

This will be a run-off of all the tools available on kili platform that aids the labeling of quality data.

**Conclusion**

**Comment by joybh98** [comment]
> As I said, the core of the article should be a concrete illustration/proof through an example that two differently labeled datasets (one with some errors, unfitted labels, etc...) can lead to a big performance difference (even with only a few errors).

I think the best approach would be to use a problem which can be understood by both beginners and expereinced practitioners, I propose to use a classification model which shows either something is happening or something is not happening. The most common application I can think of spam and not-spam. As the difference can be properly conveyed. 

If the dataset is not properly labelled, i.e spam words are not labelled as spam and vice-versa, then it can lead to detrimental change in performance of the model. Especially if it's being used in production. 

Outline:

For better perfomance of a ML model, a properly labelled dataset can make or break it.

I'll first create a classification model with a dataset that is not properly labelled, and show evaluation metrics like **accuracy**, root mean squared error, cross validation(not a metric, but it will help in driving the point home). I'll also try to test model using new data to show the reader that this model is not going to work great. Ex: I'll pass text which any human can decipher is SPAM, but the model is showing not spam because of the training data. 

Second part of the article will be focused on how using a properly labelled dataset **using the same model** shows vast difference in accuracy, and I'll show all the evaluation metrics again so the user can internalize the fact that properly labelled dataset can make or break a ML model, and it is as important as the model itself. I'll then predict values using the dataset and the user will definitely be able to see the difference.

NOTE: I'm shifting focus from showing performance difference between labelled and unlabelled data to showing performance difference between **properly labelled data** and **data that is not properly labelled**, as the issue heading states that we have to show that Better Machine Learning Demands **Better Data Labeling**

As always I'm open to any suggestions you have.

**Comment by joybh98** [comment]
@theodu , no rush. Whenever you're free let me know if the above outline is satisfactory or not.

**Comment by theodu** [comment]
Hi @joybh98 
Thank you for this update! This fits what I had in mind. Simple classification is indeed good and the spam example is easy thus illustrative for everyone.
I assign you to the article.
Whenever you can, try to use screenshots of their Kili platform, and if you need, we can provide you with a paid license to use our API ;)

**Comment by joybh98** [comment]
> Whenever you can, try to use screenshots of their Kili platform, and if you need, we can provide you with a paid license to use our API ;)

Sure will let you know !

**Comment by theodu** [comment]
Hi Joy, I well received your doc, we will review it asap
